pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1648,6 +1740,84 @@ func (gce *GCECloud) ListClusters() ([]string, error) {
 	return result, nil
 }
 
+func (gce *GCECloud) ListClusters() ([]string, error) {
+	allClusters := []string{}
+
+	for _, zone := range gce.managedZones {
+		clusters, err := gce.listClustersInZone(zone)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: Scoping?  Do we need to qualify the cluster name?
+		allClusters = append(allClusters, clusters...)
+	}
+
+	return allClusters, nil
+}
+
 func (gce *GCECloud) Master(clusterName string) (string, error) {
 	return "k8s-" + clusterName + "-master.internal", nil
 }
+
+type gceInstance struct {
+	Zone  string
+	Name  string
+	ID    uint64
+	Disks []*compute.AttachedDisk
+}
+
+type gceDisk struct {
+	Zone string
+	Name string
+	Kind string
+}
+
+func (gce *GCECloud) resolveHosts(hostNames []string) ([]*gceInstance, error) {
+	// TODO: More efficient implementation
+	// I think a good implementation might be:
+	//  * pass multiple instances to each zone (see computeHostTags)
+	//  * check each zone in turn (in parallel?)
+	//  * try local zone first (so we don't pay the tax unless we have to)
+	//  * remove instances from the list once we've found them
+	//  * we could cache 'last known zone' by name, but I don't know if this will be needed
```

Comment:
We can defer parallelizing, removing instances once they're found, caches etc until later, as I don't think they'll really buy us that much anyway.